core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -611,17 +571,17 @@ private static AuthSession fromTokenResponse(
       AuthSession session =
           new AuthSession(
               parent.headers(), response.token(), response.issuedTokenType(), credential);
+
+      Long expiresAtMillis = null;
       if (response.expiresInSeconds() != null) {
-        scheduleTokenRefresh(
-            client,
-            executor,
-            session,
-            startTimeMillis,
-            response.expiresInSeconds(),
-            TimeUnit.SECONDS);
+
+        expiresAtMillis = startTimeMillis + TimeUnit.SECONDS.toMillis(response.expiresInSeconds());
       } else if (session.expiresAtMillis() != null) {
-        scheduleTokenRefresh(
-            client, executor, session, session.expiresAtMillis(), TimeUnit.MILLISECONDS);
+        expiresAtMillis = session.expiresAtMillis();
```

Comment:
I think we should start to prefer the token's expiration if it exists. Can you swap the priority here?

Reply:
done